### ISSUE TITLE
WriteStartupDiagnostics logs error if there are two SqlPersistence

### ISSUE THREAD

**Issue by toroso** [Particular/NServiceBus.Persistence.Sql#349]
If you specify `UsePersistence<SqlPersistence>()` twice, `WriteStartupDiagnostics` will log the following error:

`Diagnostics entries contains duplicates. Duplicates: System.Linq.Lookup`2+Grouping[System.String,NServiceBus.StartupDiagnosticEntries+StartupDiagnosticEntry]`

(note the misformatted `ToString()`, it is missing a `grouping.Select(n => n.Key)` when logging)

This error _should_ be logged if you use `SqlPersistence` twice for the same `StorageType`. But if you have different `StorageType`s (e.g. outbox and saga) it should not:

```
var outboxPersistence = endpointConfiguration.UsePersistence<SqlPersistence, StorageType.Outbox>();
var sagaPersistence = endpointConfiguration.UsePersistence<SqlPersistence, StorageType.Sagas>();
```

The problem is that, in `SqlPersistence`, `AddStartupDiagnosticsSection()` is invoked with the key `"NServiceBus.Persistence.Sql.SqlDialect"`, that is, not including `StorageType`.

**Comment by SeanFeldman** [comment]
@toroso thank you for raising the issue. Are you completely blocked by it?

**Comment by toroso** [comment]
@SeanFeldman Not a blocker since `WriteStartupDiagnostics` only logs and does not prevent the service froim starting. Mostly, it's annoying to see an error in our log files.

**Comment by DavidBoike** [comment]
@toroso Thanks for the report! I've triaged this as a bug, but since it's very low impact (just annoying) and only involves the specific combination of SQL Persistence and using the different storage types.

As a result of the triage, it's considered low priority, and will be handled at some point by our maintenance release process.

**Comment by andreasohlund** [comment]
>(note the misformatted ToString(), it is missing a grouping.Select(n => n.Key) when logging)

I've confirmed that the issue with the logging has been fixed in the latest version of NServiceBus (7.2.3)

**Comment by andreasohlund** [comment]
I've raised https://github.com/Particular/NServiceBus.Persistence.Sql/pull/449 to address this. Target release will be `5.1.0`, no date set yet.
